dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

```diff
@@ -1259,76 +1277,98 @@ public void shouldReturnParserContainerUUID(final Container container, final Tem
+                { currentHost, currentHost, anotherHost, absolutePath, advanceTemplate, true},
+                { currentHost, anotherHost, anotherHost, absolutePath, advanceTemplate, true},
+                { currentHost, currentHost, defaultHost, absolutePath, advanceTemplate, true},
+                { currentHost, anotherHost, defaultHost, absolutePath, advanceTemplate, true},
+
+                { currentHost, currentHost, currentHost, relativePath, drawedTemplate, true},
+                { currentHost, currentHost, anotherHost, relativePath, drawedTemplate, false},
+                { currentHost, anotherHost, anotherHost, relativePath, drawedTemplate, false},
+                { currentHost, currentHost, defaultHost, relativePath, drawedTemplate, true},
+                { currentHost, anotherHost, defaultHost, relativePath, drawedTemplate, true},
+
+                { currentHost, currentHost, currentHost, absolutePath, drawedTemplate, true},
+                { currentHost, currentHost, anotherHost, absolutePath, drawedTemplate, true},
+                { currentHost, anotherHost, anotherHost, absolutePath, drawedTemplate, true},
+                { currentHost, currentHost, defaultHost, absolutePath, drawedTemplate, true},
+                { currentHost, anotherHost, defaultHost, absolutePath, drawedTemplate, true}
+        };
     }
 
     /**
      * Method to test: {@link com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetRenderedAPIImpl#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
-     * Given Scenario: Create a page and a advance template using a File Container in another site
+     * Given Scenario:
+     *  - Template, FileContainer and Page in the same site or different site
+     *  - Using Advance Template or not Advance Template
+     *  - Using relative or absolute path
      * ExpectedResult: should work
      *
      * @throws Exception
      */
     @Test
-    public void shouldRenderUsingOtherSiteContainer() throws Exception {
-        final Host defaultHost = APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), true);
-        final FileAssetContainer container = createFileContainer(defaultHost);
-        final Template template = new TemplateDataGen().title("PageContextBuilderTemplate"+System.currentTimeMillis())
-                .host(site)
-                .withContainer("//" + defaultHost.getName()  + container.getPath(), UUID)
-                .nextPersisted();
+    @UseDataProvider("fileContainerCases")
+    public void shouldRenderTemplateAndContainers(
+            final Host pageHost,
+            final Host templateHost,
+            final Host containerHost,
+            final Function2<FileAssetContainer, Host, String> pathConverter,
+            final Function2<Host, String, Template> templateCreator,
+            final boolean shouldWork)
+            throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6034685)